src/Dictionaries/HashedDictionaryParallelLoader.h

```diff
@@ -87,8 +95,28 @@ class HashedDictionaryParallelLoader : public boost::noncopyable
 
         for (size_t shard = 0; shard < shards; ++shard)
         {
-            if (!shards_queues[shard]->push(std::move(shards_blocks[shard])))
-                throw Exception(ErrorCodes::LOGICAL_ERROR, "Could not push to shards queue #{}", shard);
+            const auto & current_block = shards_blocks[shard];
+            while (!shards_queues[shard]->tryPush(current_block, /* milliseconds= */ 100))
```

Comment:
if I understood it correctly, we will fail the whole load operation if even a single block took more than 100ms to insert it. it seems to restrictive. and probably there are some sources that produce big blocks or a single block for the whole output.
maybe we could make it configurable or just set to a much bigger value. like 10s or a minute?

Reply:
We retry `tryPush` until block is inserted.
I replaced `push` to `tryPush` and `if` to `while`, because I don't want thread to be blocked without checking if workers are alive. So, if we cannot push, we will check status of workers and `loading_timeout` each 100 ms. If everything is alright we will retry push, otherwise we will have a chance to stop.
It's also the reason why `std::move` is not used anymore, I hope it won't be significant.